### PR TITLE
chore(db_engine_specs): clean up column spec logic and add tests

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY
Currently we're checking temporal types based on the `TemporalType` enum. This implicitly assumes that the native temporal datatype on all databases is one of the following:
```
"DATE"
"DATETIME"
"SMALLDATETIME"
"TEXT"
"TIME"
"TIME WITH TIME ZONE"
"TIMESTAMP"
"TIMESTAMP WITH TIME ZONE"
```

This is obviously not an exhaustive list. For instance, Snowflake has a datatype `TIMESTAMP_NTZ` which is not covered (see a recent hotfix for this: #22872). Luckily we already have a method `BaseEngineSpec.get_column_spec` which makes it possible to infer the correct types based on the native type that is backed by a set of overridable regexes. This matching logic correctly identifies `TIMESTAMP_NTZ`, along with a host of other similar types. See this code for details: https://github.com/apache/superset/blob/8226110be8bea6f1b06ce02def7ea56df44e6b50/superset/db_engine_specs/base.py#L200-L316 

To leverage this logic, this PR replaces all references to `TemporalType` in `convert_dttm` with calls to `get_column_spec` to more accurately infer the SQLA type from the native column type string. To ensure this extensive refactor doesn't introduce any regressions, full test coverage is added for all specs that have a `convert_dttm` method. In addition, several legacy unit tests in the integration test suite are migrated to the unit test suite, resulting in 8 deleted legacy test suites and 12 added unit tests. This will marginally speed up CI, but more importantly, it will make it easier to improve and maintain the associated logic going forward, as we now have comprehensive test coverage for this critical piece of code.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/db_engine_specs/athena.py [hunk 1]
```diff
@@ -66,10 +66,12 @@ class AthenaEngineSpec(BaseEngineSpec):
     def convert_dttm(
         cls, target_type: str, dttm: datetime, db_extra: Optional[Dict[str, Any]] = None
     ) -> Optional[str]:
-        tt = target_type.upper()
-        if tt == utils.TemporalType.DATE:
+        column_spec = cls.get_column_spec(target_type)
+        sqla_type = column_spec.sqla_type if column_spec else None
```
[the remark this answers is not in the document]
Comment (villebro) [reply superset/db_engine_specs/athena.py:70]: That's a good idea 👍 I'll do that

File: tests/unit_tests/db_engine_specs/test_presto.py [hunk 2]
```diff
@@ -46,9 +53,32 @@
...
+@pytest.mark.parametrize(
+    "native_type,sqla_type,attrs,generic_type,is_dttm",
+    [
+        ("varchar(255)", types.VARCHAR, {"length": 255}, GenericDataType.STRING, False),
+        ("varchar", types.String, None, GenericDataType.STRING, False),
+        ("char(255)", types.CHAR, {"length": 255}, GenericDataType.STRING, False),
+        ("char", types.String, None, GenericDataType.STRING, False),
+        ("integer", types.Integer, None, GenericDataType.NUMERIC, False),
+        ("time", types.Time, None, GenericDataType.TEMPORAL, True),
+        ("timestamp", types.TIMESTAMP, None, GenericDataType.TEMPORAL, True),
```
Comment (dpgaspar) [inline tests/unit_tests/db_engine_specs/test_presto.py:72]: `date` etc? do these mean superset will only support these Presto types?
prob a silly question but do these come from: https://github.com/dropbox/PyHive/blob/master/pyhive/sqlalchemy_presto.py#L30?

**Review by dpgaspar — APPROVED** [review]
Awesome work!

File: tests/unit_tests/db_engine_specs/test_presto.py (hunk 2 above)
Comment (villebro) [reply tests/unit_tests/db_engine_specs/test_presto.py:72]: I only ported over the current tests from the old `presto_tests.py` which apparently didn't have a test for `DATE`. However, notice that we're testing for it in `convert_dttm` above, so it's working at least. Once this is merged I'm hoping I can do a few more rounds of improving coverage.

**Pull request merged**